### DISCUSSION TITLE
Confusion over reference coordinate system in `massbreakdown`

### DISCUSSION THREAD

**Discussion by MarkusZimmerDLR** [DLR-SL/CPACS#832] [Q&A] [answered]
Dear CPACS folks,

### Problem description:
1. `parentUID`: we are one of those ominous institutes that use this parentUID reference as a link to the geometric components and absolutely need it. We use it to generate detailed FEM mass models.

2. One of oure colleagues calculates its masses and moments of inertia in the global KS and would like to specify them in the massbreakdonw the same way.

### There has been some confusion about this:

1. If the parentUID is specified, is the reference coordinate system for the mass entry then that of the geometric element linked via `parentUID` ?
2. `absLocal` / `absGlobal` attributes: `absLocal` means "defined in the reference coordinate system"?  `absGlobal` "defined in the global coordinate system, independent of the reference chain"?
3. Does `absGlobal` cancel out the specification of the `parentUID` for the definition of the reference coordinate system? Ergo, what is the logical hierarchy here?
4. The moment of inertia is always calculated around the specified centre of gravity?
5. The moment of inertia rotates with the reference coordinate system around the centre of gravity?

Cheers,
Markus

**Comment by MarAlder** [comment, marked as answer]
Hi Markus,

thanks for your question. I'll try to give an answer based on [point 3.2 of the CPACS documentation](https://dlr-sl.github.io/cpacs-website/documentation/CPACS_3_5_0_Docs/html/89b6a288-0944-bd56-a1ef-8d3c8e48ad95.htm):

![grafik](https://github.com/DLR-SL/CPACS/assets/43143741/f2fdf3f0-e8de-489c-bf43-8be492c58c75)

and the [`genericMassType`](https://dlr-sl.github.io/cpacs-website/documentation/CPACS_3_5_0_Docs/html/001ce50c-a53d-f47d-a985-2212a3a097fb.htm):

![grafik](https://github.com/DLR-SL/CPACS/assets/43143741/a75f6eb8-4e70-46a9-8023-b52e5eb9278f)


>     1. `parentUID`: we are one of those ominous institutes that use this parentUID reference as a link to the geometric components and absolutely need it. We use it to generate detailed FEM mass models.

Sure, just do so. You could use `componentUID` in cases where `parentUID` differs from your geometric component.


>     2. One of oure colleagues calculates its masses and moments of inertia in the global KS and would like to specify them in the massbreakdonw the same way.

Ok, fine. So `<location refType="absGlobal">`.

>     1. If the parentUID is specified, is the reference coordinate system for the mass entry then that of the geometric element linked via `parentUID` ?

Yes.

>     2. `absLocal` / `absGlobal` attributes: `absLocal` means "defined in the reference coordinate system"?  `absGlobal` "defined in the global coordinate system, independent of the reference chain"?

Yes

>     3. Does `absGlobal` cancel out the specification of the `parentUID` for the definition of the reference coordinate system? Ergo, what is the logical hierarchy here?

Yes. Similar to `transformation` as described under point 3.2 in the documentation.

>     4. The moment of inertia is always calculated around the specified centre of gravity?

I think so, but I would like to let this be confirmed by some strucutral experts.

>     5. The moment of inertia rotates with the reference coordinate system around the centre of gravity?

Not sure if I understand correctly. I think it all depends on how the coordinate systems are clustered via `parentUID`s.

Maybe some additions/corrections from @sdeinert, @sfreund-DLR, @dlr-bt?

**Reply by MarkusZimmerDLR** [reply]
Thank you for pointing at the right direction in the user guide.

Regarding the latter, there is a lot of information bundled under 
`CPACS Documentaiton > Elements > cpacs Elements > Remarks`.

From a user perspective, this is kind of hidden away. And - at least in the web version of the documentation - it lacks a proper index.
Additionally, it is also mixed with the changelog of CPACS as a whole. So, at least to me, it was not the obvious place to look for the topic I was addressing.

Maybe a reference from the documentation of the `parentUID`-node to the chapter you were linking could resolve this issue.

Thanks again.
